packages/flutter_tools/lib/src/build_system/targets/web.dart

```diff
@@ -95,9 +97,30 @@ class WebEntrypointTarget extends Target {
     // the web_plugin_registrant.dart file alongside the generated main.dart
     const String generatedImport = 'web_plugin_registrant.dart';
 
+
+    final String? iconTreeShakerFlag = environment.defines[kIconTreeShakerFlag];
+    final AssetBundle assetBundle = AssetBundleFactory.defaultInstance(
+      logger: environment.logger,
+      fileSystem: environment.fileSystem,
+      platform: environment.platform,
+    ).createBundle();
+    final int resultCode = await assetBundle.build(
+      manifestPath: environment.projectDir.childFile('pubspec.yaml').path,
+      packagesPath: environment.projectDir.childFile('.packages').path,
+      deferredComponentsEnabled: environment.defines[kDeferredComponents] == 'true',
```

Comment:
nit: remove?

Reply:
Do you mean all four of these lines? Don't we need to call `build` to generate the asset manifest?

Reply:
deferredComponentsEnabled

Reply:
Oh yeah, I think we can remove this since deferred components assets are written to the asset manifest the same way any other asset is. Addressed with commit [simplify call to assetBundle.build](https://github.com/flutter/flutter/pull/130419/commits/f504dce606fffe5f0455c52cecf0e9a9c98a3260)